ads/google/a4a/traffic-experiments.js

```diff
@@ -67,39 +67,47 @@ const INTERNALLY_SELECTED_ID = '2088462';
-    // randomlySelectUnsetPageExperiments won't add new IDs if
-    // maybeSetExperimentFromUrl has already set something for this
-    // experimentName.
-    randomlySelectUnsetPageExperiments(win, experimentInfo);
-    if (isExperimentOn(win, experimentName)) {
-      // Page is selected into the overall traffic experiment.
-      const selectedBranch = getPageExperimentBranch(win, experimentName);
-      addExperimentIdToElement(selectedBranch, element);
-      // Detect how page was selected into the overall experimentName.
-      if (isSetFromUrl) {
-        addExperimentIdToElement(EXTERNALLY_SELECTED_ID, element);
-      } else {
-        // Must be internally selected.
-        addExperimentIdToElement(INTERNALLY_SELECTED_ID, element);
-      }
-      // Detect whether page is on the "experiment" (i.e., use A4A rendering
-      // pathway) branch of the overall traffic experiment or it's on the
-      // "control" (i.e., use traditional, 3p iframe rendering pathway).
-      return selectedBranch == internalBranches.experiment ||
-          selectedBranch == externalBranches.experiment ||
-          selectedBranch == MANUAL_EXPERIMENT_ID;
+  if (!isGoogleAdsA4AValidEnvironment(win)) {
+    // Serving location doesn't qualify for A4A treatment
+    return false;
+  }
+
+  const isSetFromUrl = maybeSetExperimentFromUrl(win, element,
+      experimentName, externalBranches.control,
+      externalBranches.experiment, MANUAL_EXPERIMENT_ID);
+  const experimentInfo = {};
+  experimentInfo[experimentName] = internalBranches;
+  // Note: Because the same experimentName is being used everywhere here,
+  // randomlySelectUnsetPageExperiments won't add new IDs if
+  // maybeSetExperimentFromUrl has already set something for this
+  // experimentName.
+  randomlySelectUnsetPageExperiments(win, experimentInfo);
+  if (isExperimentOn(win, experimentName)) {
+    // Page is selected into the overall traffic experiment.
+    // In other words, if A4A has not yet launched serve A4A Fast Fetch,
+    // else serve Delayed Fetch.
+    const selectedBranch = getPageExperimentBranch(win, experimentName);
+    addExperimentIdToElement(selectedBranch, element);
+    // Detect how page was selected into the overall experimentName.
+    if (isSetFromUrl) {
+      addExperimentIdToElement(EXTERNALLY_SELECTED_ID, element);
+    } else {
+      // Must be internally selected.
+      addExperimentIdToElement(INTERNALLY_SELECTED_ID, element);
     }
+    // Detect whether page is on the "experiment" (i.e., use A4A rendering
+    // pathway) branch of the overall traffic experiment or it's on the
+    // "control" (i.e., use traditional, 3p iframe rendering pathway).
+    const selected = selectedBranch == internalBranches.experiment ||
+                     selectedBranch == externalBranches.experiment ||
+                     selectedBranch == MANUAL_EXPERIMENT_ID;
+    return (selected == !hasLaunched(win, element));
```

Comment:
If the return value is true, we should serve a Fast Fetch ad. If false, we should serve a Delayed Fetch ad. So:
selected = false, hasLaunched() = false -> the "==" is not satisfied, so we return false and serve a Delayed Fetch ad. The reasoning for this is that if A4A hasn't launched, and we're not selected into the experiment, then we shouldn't serve an A4A ad.